cmd/thanos/check.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"io/ioutil"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	thanosrule "github.com/improbable-eng/thanos/pkg/rule"
+	"github.com/oklog/run"
+	"github.com/opentracing/opentracing-go"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/pkg/rulefmt"
+	"github.com/prometheus/tsdb"
+	"gopkg.in/alecthomas/kingpin.v2"
+	"gopkg.in/yaml.v2"
+)
+
+func registerChecks(m map[string]setupFunc, app *kingpin.Application, name string) {
+	cmd := app.Command(name, "Linting tools for Thanos")
+	registerRulesCheck(m, cmd, name)
+	return
+}
+
+func registerRulesCheck(m map[string]setupFunc, root *kingpin.CmdClause, name string) {
+	checkRulesCmd := root.Command("rules", "Check if the rule files are valid or not.")
+	ruleFiles := checkRulesCmd.Arg(
+		"rule-files",
+		"The rule files to check.",
+	).Required().ExistingFiles()
+
+	m[name+" rules"] = func(g *run.Group, logger log.Logger, reg *prometheus.Registry, _ opentracing.Tracer, _ bool) error {
+		// Dummy actor to immediately kill the group after the run function returns.
+		g.Add(func() error { return nil }, func(error) {})
+		return checkRulesFiles(logger, ruleFiles)
+	}
+}
+
+func checkRulesFiles(logger log.Logger, files *[]string) error {
+	failed := tsdb.MultiError{}
+
+	for _, f := range *files {
+		n, errs := checkRules(logger, f)
+		if errs.Err() != nil {
+			level.Error(logger).Log("result", "FAILED")
+			for _, e := range errs {
+				level.Error(logger).Log("error", e.Error())
+				failed.Add(e)
+			}
+			level.Info(logger).Log()
+			continue
+		}
+		level.Info(logger).Log("result", "SUCCESS", "rules found", n)
+	}
+	if failed.Err() != nil {
+		return errors.New(failed.Error())
```

Comment:
```suggestion
		return failed
```

I think you forgot to do this one :P 
Or do you disagree?

Reply:
sorry should be there again